### PR TITLE
total count on left menu for each buy/sell according to status

### PR DESCRIPTION
**Pull request by vikas-cis**

### CODE REVIEW

File: src/app/core/market/order-status-notifier/order-status-notifier.service.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    if (orders.orders.length === 0) {
+      return;
+    }
+
+    if (!this.oldOrders || this.oldOrders.allOrders.length === 0) {
+      this.oldOrders = orders;
+    } else {
+      this.oldOrders.allOrders.forEach(order => {
+        orders.allOrders.forEach(newOrder => {
+          if (
```
Comment (kewde) [inline src/app/core/market/order-status-notifier/order-status-notifier.service.ts]: move compare function into a different function.

File: src/app/core/market/api/bid/bid.model.ts [hunk 2]
```diff
@@ -27,25 +30,52 @@ export class Bid {
 
   setFilter() {
     this.orderItems = [];
-    const orders = [];
+    this.buyOrders = [];
```
Comment (kewde) [inline src/app/core/market/api/bid/bid.model.ts]: build a new model for all this shizzle; this is starting to differentiate too much from the market api.

File: src/app/core/market/order-status-notifier/order-status-notifier.service.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+  compareOrders(newOrders: any) {
+    this.oldOrders.forEach(order => {
+      newOrders.forEach(newOrder => {
+        if (
+          order.id === newOrder.id &&
+          order.messages.action_button !== newOrder.messages.action_button &&
+          this.actionStatus.includes(newOrder.messages.action_button)
+        ) {
+          this.notifyNewStatus(newOrder);
```
Comment (kewde) [inline src/app/core/market/order-status-notifier/order-status-notifier.service.ts]: This is not part of the compare function..
Put the code from the if() statement in it.

**Comment by AllienWorks** [comment]
Tested, works fine. Not sure if it's just me, but the overall client now feels quite slow – is it because of this PR?

**Comment by vikas-cis** [comment]
@AllienWorks can you please test the PR for the messages that we discussed for notification(Order status Actions).

Thanks

**Comment by AllienWorks** [comment]
@vikas-cis fixed the styling.

Can we hide the tags:
1. until they are loaded
2. if they show only `0`?

Currently it takes about 11 sec. to load the correct numbers. It's a bit confusing.

File: src/app/core-ui/main/main-view.module.ts [hunk 4]
```diff
@@ -28,11 +29,13 @@ import { TimeoffsetComponent } from './status/timeoffset/timeoffset.component';
   ],
   exports: [
     MainViewComponent,
-    PercentageBarComponent
+    PercentageBarComponent,
+    // OrderCountComponent
```
Comment (kewde) [inline src/app/core-ui/main/main-view.module.ts]: rm

File: src/app/core-ui/main/order-count/order-count.component.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  constructor(
+    private marketState: MarketStateService,
+    private profileService: ProfileService,
+    private _orderStatusNotifierService: OrderStatusNotifierService
+  ) {}
+
+  ngOnInit() {
+    this.marketState.observe('bid')
+      .takeWhile(() => !this.destroyed)
+      .distinctUntilChanged()
```
Comment (kewde) [inline src/app/core-ui/main/order-count/order-count.component.ts]: rm or use proper deep compare filter

File: src/app/core/market/order-status-notifier/order-status-notifier.service.spec.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+import { OrderStatusNotifierService } from './order-status-notifier.service';
+import { CoreModule } from 'app/core/core.module';
+
+describe('OrderStatusNotifierService', () => {
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        HttpClientModule,
+        CoreModule.forRoot(),
+        // RpcModule.forRoot()
```
Comment (kewde) [inline src/app/core/market/order-status-notifier/order-status-notifier.service.spec.ts]: rm

**Comment by vikas-cis** [comment]
@kewde currently we don't have marketplace command to count the orders count so once we have pagination or stuff going then we can request from ludx to have that command too :)

**Pull request merged**